scripts/lis_hyperv_platform/main.ps1

```diff
@@ -14,36 +12,88 @@ param(
 $ErrorActionPreference = "Stop"
 
 $scriptPath = Split-Path -Parent $MyInvocation.MyCommand.Definition
-$JobPath = Split-Path -Parent $VHDPath
-
 . "$scriptPath\retrieve_ip.ps1"
 
-& "$scriptPath\setup_env.ps1" $JobPath $VHDPath $UserdataPath $KernelURL $InstanceName $MkIsoFS
-if ($LastExitCode -ne 0) {
-    throw
+function MyTest-Path {
+    param(
+        [String] $Path
+    )
+    if (!(Test-Path $Path)) {
+       throw "Path $Path not found"
+    }
+
 }
 
-$ip = Get-IP $InstanceName $VMCheckTimeout
-if ($ip) {
-    $throtleTimeStep = 5
-    $retryPeriod = 10
-    $retryTimes = 10
-    while ($retryTimes -gt 0) {
-        Write-Host "Trying to connect via SSH to $ip..."
-        & ssh.exe -tt -o StrictHostKeyChecking=no -i "$JobPath\$InstanceName-id-rsa" ubuntu@$ip
-        if ($LastExitCode) {
-            Write-Host "Failed to connect to $ip with error code: $LastExitCode"
-        }
-        $retryPeriod += $throtleTimeStep
-        $retryTimes = $retryTimes - 1
-        Start-Sleep $retryPeriod
+function Prepare-LocalEnv {
+    param(
+        [String] $SharedStoragePath,
+        [String] $JobId
+    )
+
+    $path = "/var/lib/lava/dispatcher/tmp/$JobId"
+    $remotePath = "H:\$JobId"
+    $localPath = "C:$path"
+
+    $SharedStoragePath = $SharedStoragePath.Replace("\\", "\")
+
+    net use H: $SharedStoragePath /persistent:NO 2>&1 | Out-Null
+    if ($LastExitCode) {
+        throw
     }
-} else {
-    throw "IP for instance $InstanceName not exposed."
+
+
+    MyTest-Path $remotePath
+
+    New-Item -Path $localPath -ItemType "directory" | Out-Null
+    Copy-Item -Path "$remotePath/*" -Destination $localPath -Force -Recurse
+
+    $localVHDPath = (Get-ChildItem -Filter "ubuntu-cloud.vhdx" -Path $localPath -Recurse ).FullName
+    MyTest-Path $localVHDPath
+
+    $lavaToolDisk = (Get-ChildItem -Filter "lava-guest.vhdx" -Path $localPath -Recurse ).FullName
+    MyTest-Path $lavaToolDisk
+
+    $remoteVHDPath = (Get-ChildItem -Filter "ubuntu-cloud.vhdx" -Path $remotePath -Recurse ).FullName
+    $remotePath = Split-Path -Parent $remoteVHDPath
+
+    return @($localVHDPath, $lavaToolDisk, $remotePath)
+}
+
+function Expand-URL {
+    param(
+        [String] $KernelUrl,
+        [String] $KernelVersion
+    )
+
+    $kernelURLExpanded = @()
+    $kernelURLExpanded += "{0}/hyperv-daemons_{1}_amd64.deb" -f @($KernelURL, $KernelVersion)
```

Comment:
Add this formatted strings to a constant array.
Something like:

KERNEL_ARTIFACTS_URL = @("{0}/hyperv-daemons_{1}_amd64.deb",
                                                 {0}/linux-headers-{1}_{1}-10.00.Custom_amd64.deb", etc)
and use a for loop here to create the actual URLs.